src/sema/namespace.rs

```diff
@@ -760,10 +760,10 @@ impl Namespace {
         id: &pt::Expression,
         diagnostics: &mut Diagnostics,
     ) -> Result<Type, ()> {
-        fn resolve_dimensions(
-            ast_dimensions: &[Option<(pt::Loc, BigInt)>],
-            diagnostics: &mut Diagnostics,
-        ) -> Result<Vec<ArrayLength>, ()> {
+        let is_substrate = self.target.is_substrate();
```

Comment:
Why is this variable necessary?

Reply:
I need this bool inside the closure. But we can't borrow `self` in the closure, because it is borrowed later in the function as well